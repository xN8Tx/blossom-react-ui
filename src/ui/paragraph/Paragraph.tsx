import React from 'react';

import type { ParagraphPropsType } from './Paragraph.types';

import style from './Paragraph.module.scss';

export const Paragraph = ({
  children,
  size,
  weight = 'regular',
  color = 'primary',
  ...props
}: ParagraphPropsType) => {
  const className = `${style.Paragraph} ${style[size]} ${style[weight]} ${style[color]}`;

  return (
    <p className={className} {...props}>
      {children}
    </p>
  );
};
