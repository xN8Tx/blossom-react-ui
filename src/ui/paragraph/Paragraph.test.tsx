import React from 'react';

import { cleanup, render, screen } from '@testing-library/react';

import { Paragraph } from './Paragraph';

describe('Paragraph', () => {
  afterEach(() => cleanup());
  test('Should render', () => {
    render(
      <Paragraph color='primary' size='xs' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toBeInTheDocument();
  });

  it('Snapshot XS', () => {
    render(
      <Paragraph color='primary' size='xs' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toMatchSnapshot();
  });

  it('Snapshot S', () => {
    render(
      <Paragraph color='primary' size='s' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toMatchSnapshot();
  });

  it('Snapshot M', () => {
    render(
      <Paragraph color='primary' size='m' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toMatchSnapshot();
  });

  it('Snapshot L', () => {
    render(
      <Paragraph color='primary' size='l' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toMatchSnapshot();
  });

  it('Snapshot XL', () => {
    render(
      <Paragraph color='primary' size='xl' weight='regular'>
        Hello world!
      </Paragraph>,
    );

    const button = screen.getByText('Hello world!');

    expect(button).toMatchSnapshot();
  });
});
