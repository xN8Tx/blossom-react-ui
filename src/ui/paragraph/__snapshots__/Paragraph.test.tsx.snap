// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Snapshot L 1`] = `
<p
  class="Paragraph l regular primary"
>
  Hello world!
</p>
`;

exports[`Paragraph Snapshot M 1`] = `
<p
  class="Paragraph m regular primary"
>
  Hello world!
</p>
`;

exports[`Paragraph Snapshot S 1`] = `
<p
  class="Paragraph s regular primary"
>
  Hello world!
</p>
`;

exports[`Paragraph Snapshot XL 1`] = `
<p
  class="Paragraph xl regular primary"
>
  Hello world!
</p>
`;

exports[`Paragraph Snapshot XS 1`] = `
<p
  class="Paragraph xs regular primary"
>
  Hello world!
</p>
`;
